### ISSUE TITLE
Reconnection fails after prolonged Internet interruption

### ISSUE THREAD

**Issue by piddyt-fambyt** [mneudert/instream#63]
Hi!
Thanks for this repository. It enabled me to switch to Elixir for my little project.
In my application I regularly want to write data to Influx with an intermittent internet connection.
To address this I use `Connection.status` to see if there is currently Internet available.
If :error is returned data is buffered, otherwise data is sent. In my test scenario this was working just fine.
However if internet is not available for a longer time (this means tens or hundreds of failed `Connection.status` queries), it won't ever reconnect, bus returns constantly :error. After a complete restart of the application it works fine again.
Unfortunately I am no elixir expert and don't really now how I could see why this happens.

**Comment by mneudert** [comment]
The problem is most likely based on `:hackney` not removing connections from its pool after encountering a network error. Using the builtin pool stat function I can verify that with an intentionally unreachable connection endpoint:

```elixir
iex(1)> :hackney_pool.get_stats(:default)
[name: :default, max: 50, in_use_count: 0, free_count: 0, queue_count: 0]
iex(2)> Connection.ping()
:error
iex(3)> :hackney_pool.get_stats(:default)
[name: :default, max: 50, in_use_count: 1, free_count: 0, queue_count: 0]
```

However two days ago a fix was released with `:hackney` version `1.17.0` so these errors do not drain the connection pool anymore (above example stays as `in_use_count: 0`).

If you are able to update your dependencies (`mix deps.update hackney` if you don't want to update everything) the problem should solve itself if I am not mistaken here.

**Comment by piddyt-fambyt** [comment]
Thanks for the fast reply. I updated hackney and it looks good so far!
